common:
  img_topic: "/left_camera/image"
  lid_topic: "/livox/lidar"
  imu_topic: "/livox/imu"
  img_en: 1
  lidar_en: 1
  ros_driver_bug_fix: false

extrin_calib:
  extrinsic_T: [0.04165, 0.02326, -0.0284]
  extrinsic_R: [1, 0, 0,
                0, 1, 0,
                0, 0, 1]
  Rcl: # PackBOT
      [ 0.0062095,  -0.999969,    0.00476795,
       -0.00404929, -0.00479315, -0.99998,
       0.999973,    0.00619007, -0.00407893]
  Pcl: [-0.0451984,-0.00817436, 0.0267529 ]

time_offset:
  imu_time_offset: 0.0
  img_time_offset: 0.000
  exposure_time_init: 0.03

preprocess:
  point_filter_num: 1
  filter_size_surf: 0.05
  lidar_type: 1 # Livox Avia LiDAR
  scan_line: 6
  blind: 1
  inf_bound: 70

vio:
  max_iterations: 10
  outlier_threshold: 1000 # 78 100 156 #100 200 500 700 infinite
  img_point_cov: 100 # 100 1000 
  patch_size: 8
  patch_pyrimid_level: 5
  normal_en: true
  raycast_en: true
  inverse_composition_en: false
  exposure_estimate_en: true
  inv_expo_cov: 0.1

imu:
  imu_en: true
  imu_int_frame: 200
  acc_cov: 1 # 0.2
  gyr_cov: 0.5 # 0.5
  b_acc_cov: 0.0001 # 0.1
  b_gyr_cov: 0.0001 # 0.1

lio:
  max_iterations: 10
  dept_err: 0.02
  beam_err: 0.05
  min_eigen_value: 0.0025 # 0.005
  voxel_size: 1
  max_layer: 6
  max_points_num: 100
  layer_init_num: [5, 5, 5, 5, 5]

local_map:
  map_sliding_en: true
  half_map_size: 50
  sliding_thresh: 8

uav:
  imu_rate_odom: false
  gravity_align_en: true

publish:
  dense_map_en: true
  pub_effect_point_en: false
  pub_plane_en: false
  pub_scan_num: 1
  blind_rgb_points: 0.0

evo:
  seq_name: "CBD_Building_01"
  pose_output_en: false

pcd_save:
  pcd_save_en: true
  colmap_output_en: false # need to set interval = -1
  filter_size_pcd: 0.01
  interval: -1
  # how many LiDAR frames saved in each pcd file;
  # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.
